### PR TITLE
Llvm 3.8 travis

### PR DESCRIPTION
**Pull request by cocreature**
This travis config drops support for 7.6 and enables travis for 8.0, i.e. the last three GHC releases.

The build fails (you wanted small PRs so I didn’t include the commits that fix it)

All three PRs need the metadata PR to pass. 8.0 needs the PR that provides the necessary updates, 7.10 seems to need the changes to the flag selection to pass. ftr my `llvm-3.8` branch uses the exact same travis config and all tests pass.

### CODE REVIEW

File: cabal.project [hunk 1]
```diff
@@ -0,0 +1,2 @@
+packages: ./llvm-general/
+          ./llvm-general-pure/
```
Comment (bscarlet) [inline cabal.project]: Do you really mean to omit the trailing newline?

**Comment by bscarlet** [comment]
Aside from the trailing newline question, looks good.

**Comment by cocreature** [comment]
Trailing newline added.

**Comment by bscarlet** [comment]
For extra credite, now cherry pick this change into the llvm-3.4 and llvm-3.5 branches and adjust it however is necessary.

**Pull request merged**